src/app/workbasket/workbasket-input-filter.service.ts

```diff
@@ -17,11 +19,19 @@ export class WorkbasketInputFilterService {
 
   getWorkbasketInputs(jurisdictionId: string, caseTypeId: string): Observable<WorkbasketInputModel[]> {
     let url = this.getWorkbasketInputUrl(jurisdictionId, caseTypeId);
+    this.currentJurisdiction = jurisdictionId;
+    this.currentCaseType = caseTypeId;
     return this.httpService
       .get(url)
       .map(response => {
         let workbasketInputs = response.json();
-        workbasketInputs.forEach( item => { item.field.label = item.label; });
+        if (!(this.currentJurisdiction !== jurisdictionId || this.currentCaseType !== caseTypeId)) {
```

Comment:
Is it not simpler to write:
`this.currentJurisdiction === jurisdictionId && this.currentCaseType === caseTypeId`
and maybe wrap it in a method `hasRequiredData` or the like?